# finding_loci_based_off_primers
Uses BLAST and R scripts to find genomic position of loci based off their primer sequences and expected size

### Things you need
1. A genome in fasta format
2. A tab/space-delimited list of your primers with locus name in the 1st column, F primer (5' to 3') in the 2nd column, R primer (5' to 3') in the 3rd column, and expected size including primers (bp) in the 4th. e.g.
```
2F91	GCATTTCTGGGCTGTAACAT	 AAAGGACAATGTAATTGGTG 		99
4E8	TGTGGGAAACCAGAGGAAA	 CAGGGGAAAAATAGAGAGGG 		99
4H2	GTTTTGACCTACAGAGAGAG	 GATAGGACAATCAACAGGCT 			214
ADCbm	GATGTGAGTAACCAGCCACT	 ATAACACAGGAGCGGTGA 		177
Ase64	CCACCTTTCATACTGGGGAG	 TTCAGCCAGTCAGTGTAGCC 			399
ASu15	AATAGATTCAGGTGCTTTTTCC	 GGTTTTTGAGAAAATTATACTTTCAG 			124
CcaTgu21	GGCAGACATGATTGCATCC	 TCTCAGTGGTCATTGGAAAGTG 			205
DkiB119	CATACAACTTCATGACTACCATAGCAC	 TCCATAGTGACATAGAACGAGCTG 			230
Escu6	CATAGTGATGCCCTGCTAGG	 GCAAGTGCTCCTTAATATTTGG 			110
GCSW42	GGCTTCTCTGGTTGCATGTC	 ACAGTAATCCCCAGCCATCA 			215
Ind28	CCCAGGAAGTATCCCAGAA	 CCTCCAATGCTTTAGTGACC 			166
Mjg1	CCCGGGAAAGGCTTCGTCTTC	 GGAGATTTTATATCGGTGGC 			172
Pau01	TTAGAAGTGAAAGGCTTG	 GAGGAATAAAAACAATGC 			110
Pau04	AATAAAGCAGATACTGAG	 ACAGGTAAACCAGAGCAG 			131
Pau06	CTGAGGTTCAAAGTTTCC	 ACCAGCCATCCTTATGC 			102
Pau07	CTTTCCTTGACTGAAGTG	 TTAGCTTCATTTCCAGTC 			106
Pau09	ATGATGTAGTCAGAGTCG	 TATTTTGCAACCTTCTTG 			114
Pau23	CAGGGCATTTACAGATTTTCC	 CTCATCTTGCACACTGCTG 			188
Pau26	AAATTACTACAGTGTTACGGTGAAAA	 GGGACCACCAAGAACTTCAA 			170
Pau66	TGGGCCAGTTTATACCCTCT	 ATGAAAGGGTTCCATGATGC 		106
Pau67	CCAGGAAAGGTGCTCAGAGT	 TGTCTGTGTTGGCCTGATCT 			181
Pocc6	TCACCCTCAAAAACACACACA	 ACTTCTCTCTGAAAAGGGGAGC 			184
Ppi2	CACAGACCATTCGAAGCAGA	 GCTCCGATGGTGAATGAAGT 			324
TG02	TGTGTGTTGACAGTATTCTCTTGC	 TTTAAACCTAATAAACGTCACACAGTC 			260
TG11	ACAAACTAAGTACATCTATATCTGAAG	 TAAATACAGGCAACATTGG 			224
Tgu01	TGCGGTCTGTATGGAAATAGTC	 CTTGCAATACTCTCTGCCTCA 			192
Tgu03	TCTCTCTGCTAGGGATAAACAGTG	 TGCTCCCTCCCTCCAGTAAC 			158
TguEST09	AACCCAACCAACAAAATTGG	 CCAACTATCAGTTTTACAAGGCATAC 			173
```
3. BLAST and R installed
4. An idea of the tolerance around the size of locus you want to detect (e.g. for our microsatellite example, we'd expect an individual to be within 50 bp of the average locus size)


Credit to IDT (https://sg.idtdna.com/pages/education/decoded/article/tips-for-using-blast-to-locate-pcr-primers) for the idea behind this pipeline
